### ISSUE TITLE
Query on setting availability_set in Bosh V2 manifest

### ISSUE THREAD

**Issue by jagadish-kb** [cloudfoundry-incubator/bosh-azure-cpi-release#329]
Background of our project: Our project is [service broker implementation](https://github.com/openservicebrokerapi/servicebroker/blob/v2.12/spec.md) on Cloud Foundry which provisions deployments on the fly based on user requests.

What we are trying to achieve: We are migrating from Bosh V1 manifest to V2 manifest. 

Problem statement: 
We generate the deployment manifest on the fly at run time and availability set was set in each of the resource pools with a unique name and the respective resource pool was referenced in the jobs section. However as part of V2 manifest resource pools no longer exist and we need to instead define VM_TYPES and availability set as part of its cloud properties and upload the same in cloud config.

Now our requirement is to ensure that each deployment gets a unique availability set (per type of tier). Since we can no longer have availability set in the individual manifest & its part of cloud config, how can we achieve this? 
[vm_type in[ instance group ](https://bosh.io/docs/manifest-v2.html#instance-groups) is a string , so not sure how we can specify av set value from individual manifests.]

One thought was before sending in the deploy request we update cloud config vm_type with the required availability_set value in cloud_properties and then fire the deploy. However since we are a provisioning system we could at a time be creating more than one request at a time & this might conflict across deployments. So ideally its best if somehow we can specify the same in the deployment manifest itself. 

Any thoughts around how to address this?

**Comment by AbelHu** [comment]
@jagadish-kb Could you not set the avset name but let Azure CPI pick up the group name as the avset name?
BOSH generates [`env.bosh.group`](https://github.com/cloudfoundry/bosh/blob/9d507676a1ad30228ef2d81f19314396a4a6b67b/src/bosh-director/lib/bosh/director/vm_creator.rb#L171) for every pool and Azure CPI uses this as the avset name when `availability_set` is not specified.

http://bosh.io/docs/azure-cpi.html#resource-pools:
```
availability_set [String, optional]: Name of an availability set to use for VMs. More details.
If available set does not exist, it will be automatically created.
If availability_set is not specified, Azure CPI will search env.bosh.group as the name of availability set.
bosh release v258+ will generate a value for env.bosh.group automatically.
On Azure the length of the availability set name must be between 1 and 80 characters. The name got from env.bosh.group may be too long. CPI will truncate the name to the following format az-MD5-[LAST-40-CHARACTERS-OF-GROUP] if the length is greater than 80.
CPI v27+ will delete the empty availability set.
```

**Comment by jagadish-kb** [comment]
@AbelHu Our initial approach was exactly like how you have suggested above. We were not setting the av set explicitly and were letting the CPI set it. However our requirement is something like the below & the above approach was not working:
Job/Instance Group 1 : PG Pool nodes
Job/Instance Group 2: PG Master nodes
Job/.Instance Group 3: PG Slave nodes

Now we want all nodes in instance group 2 & 3 in the same availability set and Job1 nodes in another availability set. With the approach you suggested we ended up having 3 AV Set with each nodes from each of the jobs ending up in its own AV Set.  
To fix this issue, we ended up having two resource pools (`rp1` & `rp2`) defined with avset name explicitly set in `rp1` and `rp2` uniquely. `Job1` referred to resource pool `rp1` and `Job2/Job3` both referred resource pool `rp2`. This works fine in V1 manifest however not sure how such a requirement can be met in V2 manifest.
